library/psa_crypto.c

```diff
@@ -6023,16 +6023,17 @@ static psa_status_t psa_generate_derived_key_internal(
     return status;
 }
 
-static const psa_key_generation_method_t default_method = PSA_KEY_GENERATION_METHOD_INIT;
+static const psa_key_production_parameters_t default_production_parameters =
+    PSA_KEY_PRODUCTION_PARAMETERS_INIT;
 
-int psa_key_generation_method_is_default(
-    const psa_key_generation_method_t *method,
-    size_t method_data_length)
+int psa_key_production_parameters_are_default(
+    const psa_key_production_parameters_t *params,
+    size_t params_data_length)
 {
-    if (method->flags != 0) {
+    if (params->flags != 0) {
```

Comment:
Just a minor query while I am reviewing: Is there a reason to have two distinct `if` statements here and not just combine them with the condition `if (params->flags != 0 || params_data_length != 0)` ?

I wondered if that might look a bit nicer seeing as the two `if` statements appear related, in the sense that they are both testing for things that would indicate that the parameters being used are not default ones.

Reply:
I prefer to have separate `if` statements because that way you can put a breakpoint on only one of the conditions, and the conditions show up separately in line or branch coverage measurements.

(That's only if the code inside the branch is something very simple, like `return ERROR_CODE` or `goto error`. If it's more complex then I do use the `||` operator.)

Reply:
Makes absolute sense, hadn't thought about it from a debugging PoV. Thanks for explaining.